Account/FirefoxAccount.swift

```diff
@@ -140,21 +149,26 @@ public class FirefoxAccount {
             configurationLabel = configurationLabel,
             email = dictionary["email"] as? String,
             uid = dictionary["uid"] as? String {
+                let fxaDeviceId = dictionary["fxaDeviceId"] as? String
+                let deviceRegistrationVersion = dictionary["deviceRegistrationVersion"] as? Int ?? 0
                 let stateCache = KeychainCache.fromBranch("account.state", withLabel: dictionary["stateKeyLabel"] as? String, withDefault: SeparatedState(), factory: stateFromJSON)
                 return FirefoxAccount(
                     configuration: configurationLabel.toConfiguration(),
                     email: email, uid: uid,
+                    fxaDeviceId: fxaDeviceId, deviceRegistrationVersion: deviceRegistrationVersion,
                     stateCache: stateCache)
         }
         return nil
     }
 
     public enum AccountError: MaybeErrorType {
         case NotMarried
+        case DeviceRegistrationFailed
```

Comment:
Oops, I had removed this at one point, but must have accidentally reverted it. This is left over from earlier commits where we did registration in `FirefoxAccount`. It's now handled separately by `FxADeviceRegistrator`.

This reminds me of one of the main questions I had working on this PR: does it make sense to do the device registration step outside of the account marriage, or should we tack on registration as another state in the state machine? I went with the former simply because it's what we do on Android, but I don't know how much sense it makes to use an unregistered account as we adopt these new FxA APIs.